yt_dlp/extractor/floatplane.py

```diff
@@ -0,0 +1,268 @@
+
+    def _real_extract(self, url):
+        post_id = self._match_id(url)
+
+        post_data = self._download_json(
+            'https://www.floatplane.com/api/v3/content/post', post_id, query={'id': post_id},
+            note='Downloading post data', errnote='Unable to download post data')
+
+        if not any(traverse_obj(post_data, ('metadata', ('hasVideo', 'hasAudio')))):
+            raise ExtractorError('Post does not contain a video or audio track', expected=True)
+
+        items = []
+        for media in traverse_obj(post_data, (('videoAttachments', 'audioAttachments'), ...)):
+            media_id = media['id']
+            media_typ = media.get('type') or 'video'
+
+            metadata = self._download_json(
+                f'https://www.floatplane.com/api/v3/content/{media_typ}', media_id, query={'id': media_id},
+                note=f'Downloading {media_typ} metadata')
+
+            stream = self._download_json(
+                'https://www.floatplane.com/api/v2/cdn/delivery', media_id, query={
+                    'type': 'vod' if media_typ == 'video' else 'aod',
+                    'guid': metadata['guid']
+                }, note=f'Downloading {media_typ} stream data')
+
+            path_template = traverse_obj(stream, ('resource', 'uri', {str}))
+
+            def format_path(params):
+                path = path_template
+                for i, val in (params or {}).items():
+                    path = path.replace(f'{{qualityLevelParams.{i}}}', val)
+                return path
+
+            formats = []
+            for quality in traverse_obj(stream, ('resource', 'data', 'qualityLevels', ...)):
+                url = urljoin(stream['cdn'], format_path(traverse_obj(
+                    stream, ('resource', 'data', 'qualityLevelParams', quality['name']))))
+                formats.append({
+                    **traverse_obj(quality, {
+                        'format_id': 'name',
+                        'format_note': 'label',
+                        'width': ('width', {int}),
+                        'height': ('height', {int}),
+                    }),
+                    **parse_codecs(quality.get('codecs')),
+                    'url': url,
+                    'ext': determine_ext(url.partition('/chunk.m3u8')[0], 'mp4'),
+                })
+
+            items.append({
+                'id': media_id,
+                **traverse_obj(metadata, {
+                    'title': 'title',
+                    'duration': ('duration', {int_or_none}),
+                    'thumbnail': ('thumbnail', 'path'),
+                }),
+                'formats': formats,
+            })
+
+        uploader_url = format_field(traverse_obj(
+            post_data, ('creator', 'urlname')), template='https://www.floatplane.com/channel/%s/home', default=None)
+        channel_url = format_field(traverse_obj(
+            post_data, ('channel', 'urlname')), template=f'{uploader_url}/%s', default=None)
```

Comment:
We can shorten the lines a bit by passing both the object and field args
```suggestion
        uploader_url = format_field(traverse_obj(
            post_data, 'creator'), 'urlname', 'https://www.floatplane.com/channel/%s/home', default=None)
        channel_url = format_field(traverse_obj(
            post_data, 'channel'), 'urlname', f'{uploader_url}/%s', default=None)
```
&nbsp;
Maybe urljoin could actually be preferable for `channel_url` though, since with the `format_field` way we could end up with `None/channel`
```suggestion
        uploader_url = format_field(traverse_obj(
            post_data, 'creator'), 'urlname', 'https://www.floatplane.com/channel/%s/home', default=None)
        channel_url = urljoin(f'{uploader_url}/', traverse_obj(post_data, ('channel', 'urlname')))
```